NFC fixes

cstamas opened the pull request:
As it turns out, NFC is used by ALL repository types (used as populated).

Even groups, even if in their case the actual _content_ of NFC is neglected, they still use it!
Before this change, the size of NFC in case of a non serviceable request to public group would raise
with 1+groupMemberSize. After this change, size of NFC would raise with the count of affected proxy
repositories only (1 in this case, Central only).

The testing of these changes were applied to RSO with good results, hosted reposes did not make disks evaporate.

cstamas committed:
NFC is used by ALL repository types.

Even groups, even if in their case the actual _content_ of NFC is neglected, they still use it!
Before this change, the size of NFC in case of a non serviceable request to public group would raise
with 1+groupMemberSize. After this change, size of NFC would raise with the count of affected proxy
repositories only (1 in this case, Central only).

bdemers commented on nexus/nexus-configuration-model/src/main/filtered-resources/META-INF/nexus/default-oss-nexus.xml:
```diff
@@ -123,7 +123,7 @@
       <providerRole>org.sonatype.nexus.proxy.repository.Repository</providerRole>
       <providerHint>maven2</providerHint>
       <localStatus>IN_SERVICE</localStatus>
-      <notFoundCacheActive>true</notFoundCacheActive>
```
if we end up going this route, don't forget about the nexus.xml in the professional edition

bdemers commented:
seems like a good approach

cstamas commented:
This would need some verification, ie. adjusting these on RSO or OSO and see what impact it does...

cstamas committed:
NFC Experiments: covering new repository creation (over UI or programmatically)

Previous commit "fixes" static configuration and upgrade, but new repository creations
was not handled. Here, we reverted the situation, and explicitly enable NFC for new
proxy repositories only.

cstamas committed:
Added component to interfere in driven way with configuration to set NFC overrides or back out from them.

peterlynch commented:
looks wise

adreghiciu commented:
+1. BTW, we keep config version as 1.10? That does not make sense

bdemers commented:
Are we going to get this in for the next build ?

cstamas commented:
Yes, but as I said, this is complete for OSS only, we'd still need pro changes....

mcculls commented:
+1

mcculls commented on nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/rest/repositories/RepositoryListPlexusResource.java:
```diff
@@ -247,6 +247,8 @@ else if ( !RepositoryBaseResourceConverter.REPO_TYPE_GROUP.equals( resource.getR
                 appModel.getLocalStorage().setProvider( "file" );
+                
+                appModel.setNotFoundCacheActive( true );
             }
```
Probably a silly question - everywhere else NFC is set according to if repo is a proxy or not, whereas here it's set for a local repo with non-local storage - is this a special case?

cstamas replied:
:beer: X1

cstamas replied:
Fixed in subsequent commit. Thanks.

cstamas committed:
Update nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/rest/repositories/RepositoryListPlexusResource.java

bdemers commented:
+1
